lib/matplotlib/text.py

```diff
@@ -1474,7 +1474,8 @@ def __init__(self, artist, ref_coord, unit="points"):
         self.set_unit(unit)
 
     def set_unit(self, unit):
-        assert unit in ["points", "pixels"]
+        if unit not in ["points", "pixels"]:
+            raise ValueError("'unit' must be one of ['points', 'pixels']")
```

Comment:
Should probably make this consistent with the rest of the messages that ar `[ foo | bar ]`